TradePlay/src/context/auth-context.jsx

```diff
@@ -0,0 +1,36 @@
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+import { userDataReducer } from "./auth-reducer";
+
+const AuthContext = createContext();
+const useAuth = () => useContext(AuthContext);
+
+const AuthProvider = ({ children }) => {
+  const initialUserData = {
```

Comment:
Suggestion: you can pass this `initialUserData` object as a default state parameter to the reducer, and return the state in the default case of switch case.

Reply:
Yes, will do that